src/polyswarm_api/__main__.py

```diff
@@ -227,45 +177,32 @@ def search():
 
 
 @click.option('-r', '--hash-file', help='File of hashes, one per line.', type=click.File('r'))
-@click.option('--hash-type', help='Hash type to search [sha256|sha1|md5], default=sha256', default='sha256')
+@click.option('--hash-type', help='Hash type to search [default:sha256|sha1|md5]', default=None)
 @click.argument('hashes', nargs=-1)
 @search.command('hash', short_help='search for hashes separated by space')
 @click.pass_context
 def hashes(ctx, hashes, hash_file, hash_type):
     """
-    Search PolySwarm for files matching sha256 hashes
+    Search PolySwarm for files matching hashes
     """
 
-    def _get_hashes_from_file(file):
-        return [h.strip() for h in file.readlines()]
-
-    def _remove_invalid_hashes(hash_candidates, candidates_hash_type):
-
-        def is_valid_hash(hash_candidate):
-            return (candidates_hash_type == 'sha256' and _is_valid_sha256(hash_candidate)) or \
-                   (candidates_hash_type == 'sha1' and _is_valid_sha1(hash_candidate)) or \
-                   (candidates_hash_type == 'md5' and _is_valid_md5(hash_candidate))
-
-        valid_hashes = []
-        for candidate in hash_candidates:
-            if is_valid_hash(candidate):
-                valid_hashes.append(candidate)
-            else:
-                logger.warning('Invalid hash %s, ignoring.', candidate)
-        return valid_hashes
-
     api = ctx.obj['api']
 
     hashes = list(hashes)
 
     if hash_file:
-        hashes += _get_hashes_from_file(hash_file)
+        hashes += get_hashes_from_file(hash_file)
+
+    # validate default hashes [sha1|sha256|md5]
+    if not hash_type or hash_type == 'sha1' or \
```

Comment:
Same thing, there should be a global list of supported hashes somewhere so we don't have to change 5 different `if` statements if we update it.

Reply:
yep. i was thinking about same thing, but we'll have to rewrite it with the non defaults ones. Anyway i will do something with this.

Reply:
done